mitosheet/src/mito/components/SearchBar.tsx

```diff
@@ -10,71 +10,225 @@ import Input from './elements/Input';
+    scrollAndRenderedContainerDiv: HTMLDivElement | null;
+    sheetView: SheetView;
+    gridState: GridState;
 }
 
 export const SearchBar: React.FC<SearchBarProps> = (props) => {
-    const { setUIState, uiState, mitoAPI } = props;
+    const {
+        setUIState,
+        uiState,
+        mitoAPI,
+        containerDiv,
+        scrollAndRenderedContainerDiv,
+        sheetView,
+        gridState
+    } = props;
+    const { searchValue, currentMatchIndex, matches } = uiState.currOpenSearch;
+
+    // totalMatches is undefined when we're making the API call to get the total number of matches.
     const [totalMatches, setTotalMatches] = React.useState<number | undefined>(undefined);
-    const searchValue = uiState.currOpenSearch.searchValue;
+    const [ showCautionMessage, setShowCautionMessage ] = React.useState<boolean>(false);
+
+    const scrollMatchIntoView = (match?: { row: number; col: number }) => {
+        // Columns have row index -1, so we check for that first.
+        if (match?.row === -1) {
+            scrollColumnIntoView(
+                containerDiv,
+                scrollAndRenderedContainerDiv,
+                sheetView,
+                gridState,
+                match?.col
+            )
+        } else if (match !== undefined) {
+            ensureCellVisible(
+                containerDiv,
+                scrollAndRenderedContainerDiv,
+                sheetView,
+                gridState,
+                match.row,
+                match.col,
+            )
+        }
+    }
 
-    /**
-     * Update the total number of matches when the search value changes
-     */
+    // Call the backend to get the new match information when the search value or sheet index changes.
     useDebouncedEffect(() => {
-        // If the search value is empty, set the total matches to 0
+        // If the search value is empty, set the total matches to 0 and don't call the API.
         if (searchValue === undefined || searchValue === '') {
             setTotalMatches(0);
             return;
         }
+
+        // Call the API to get the total number of matches and the displayed matches. 
         void mitoAPI.getTotalNumberMatches(uiState.selectedSheetIndex, searchValue ?? '').then((response) => {
             if ('error' in response) {
                 return;
             }
-            const total = response.result;
-            if (total !== undefined && !isNaN(Number(total))) {
-                setTotalMatches(Number(total));
+            const new_total_number_matches = response.result.total_number_matches;
+            const new_matches = response.result.matches;
+            // Update the total matches. 
+            if (!isNaN(Number(new_total_number_matches))) {
+                setTotalMatches(Number(new_total_number_matches));
+            }
+
+            if (new_matches?.length > 0) {
+                scrollMatchIntoView(new_matches[currentMatchIndex]);
             }
```

Comment:
Why do we want to scroll the currentMatchIndex? Wouldn't we want to reset that?

Reply:
I added this so that it'd automatically scroll to the first match when the API call returns w/ the new matches.

Reply:
Otherwise you would have to press enter to scroll to the new match - I think different applications handle this differently so up to you if you want to keep this behavior!